lib/Frontend/ArgsToFrontendOutputsConverter.cpp

```diff
@@ -578,6 +587,7 @@ createFromTypeToPathMap(const TypeToPathMap *map) {
     return paths;
   const std::pair<file_types::ID, std::string &> typesAndStrings[] = {
       {file_types::TY_ObjCHeader, paths.ObjCHeaderOutputPath},
+      {file_types::TY_ObjCHeader, paths.CxxHeaderOutputPath},
```

Comment:
This is a problem, because we have two different outputs associated with the same entry in the output file map, so we'll end up writing both the ObjC and C++ header to the same place. I've fixed the issue via https://github.com/apple/swift/pull/41015, but I think we should reconsider having separate ObjC vs. C++ generated headers. It's a mess for the build system, and what would we do about Objective-C++? Instead, I think we should generate a single header that uses `#ifdefs` to appropriately compile as C/ObjC/C++/ObjC++.

Reply:
Hi Doug, thanks for fixing this, and sorry about the breakage. I'm not sure if we would like to generate a single header, instead we actually would prefer to generate multiple C++ headers as they will be target-specific, one per target, and then have the build system create a unified C++ header one from them. Do you think that would be a reasonable idea, and should we even support path map in that case? Also, I don't believe we want to emit Objective-C++ headers, as that would be ambiguous - we would either want to emit Objective-C **or** C++, and then the programmer can import their preferred interface into Objective-C++ as desired.